app/src/main/java/com/exallium/mvvmcleanexample/app/users/UserEditActivity.kt

```diff
@@ -40,19 +44,31 @@ class UserEditView : AppCompatActivity() {
 
         binding.viewModel = viewModel
 
-        router.transitions().subscribe {
-            when (it) {
-                is AppRouter.Transition.GoBack -> onBackPressed()
-                is AppRouter.Transition.DisplayMessage -> displayMessage(it.message)
-                is AppRouter.Transition.DisplayError -> displayError(binding.root, it.message)
-            }
+        router.transitions().subscribe(this::onNextTransition)
+    }
+
+    private fun onNextTransition(transition: AppRouter.Transition) {
+        when (transition) {
+            is AppRouter.Transition.GoBack -> finish()
```

Comment:
I dont think you are supposed to invoke `onBackPressed()` programatically, this was causing a crash for me.